client/src/main/java/com/netflix/conductor/client/http/EventClient.java

```diff
@@ -14,101 +14,61 @@
-     * @param filters Chain of client side filters to be applied per request
-     */
-    public EventClient(ClientConfig config, ClientHandler handler, ClientFilter... filters) {
-        this(config, new DefaultConductorClientConfiguration(), handler, filters);
+    public EventClient(RequestHandler requestHandler) {
+        this(requestHandler, null);
     }
 
-    /**
-     * @param config REST Client configuration
-     * @param clientConfiguration Specific properties configured for the client, see {@link
-     *     ConductorClientConfiguration}
-     * @param handler Jersey client handler. Useful when plugging in various http client interaction
-     *     modules (e.g. ribbon)
-     * @param filters Chain of client side filters to be applied per request
-     */
     public EventClient(
-            ClientConfig config,
-            ConductorClientConfiguration clientConfiguration,
-            ClientHandler handler,
-            ClientFilter... filters) {
-        super(config, clientConfiguration, handler);
-        for (ClientFilter filter : filters) {
-            super.client.addFilter(filter);
-        }
+            RequestHandler requestHandler, ConductorClientConfiguration clientConfiguration) {
+        super(requestHandler, clientConfiguration);
     }
 
     /**
-     * Register an event handler with the server
+     * Register an event handler with the server.
      *
-     * @param eventHandler the eventHandler definition
+     * @param eventHandler the eventHandler definition.
      */
     public void registerEventHandler(EventHandler eventHandler) {
         Preconditions.checkNotNull(eventHandler, "Event Handler definition cannot be null");
-        postForEntityWithRequestOnly("event", eventHandler);
+        post("event", eventHandler);
     }
 
     /**
-     * Updates an event handler with the server
+     * Updates an event handler with the server.
      *
-     * @param eventHandler the eventHandler definition
+     * @param eventHandler the eventHandler definition.
      */
     public void updateEventHandler(EventHandler eventHandler) {
         Preconditions.checkNotNull(eventHandler, "Event Handler definition cannot be null");
         put("event", null, eventHandler);
     }
 
     /**
-     * @param event name of the event
-     * @param activeOnly if true, returns only the active handlers
-     * @return Returns the list of all the event handlers for a given event
+     * @param event name of the event.
+     * @param activeOnly if true, returns only the active handlers.
+     * @return Returns the list of all the event handlers for a given event.
      */
     public List<EventHandler> getEventHandlers(String event, boolean activeOnly) {
-        Preconditions.checkArgument(
-                org.apache.commons.lang3.StringUtils.isNotBlank(event), "Event cannot be blank");
+        Preconditions.checkArgument(StringUtils.isNotBlank(event), "Event cannot be blank");
```

Comment:
not directly related to these changes, but since we are making a breaking change anyway, can we get rid of guava from the client. I believe it is transitively pulled in by the eureka client and the only usages I can see are for `Preconditions`. 
Looking at the eureka client gradle file, `guava` is a `compile` dependency which means it leaks out transitively, we can add an exclude for it on our side.

Reply:
Good point. I will see if i can exclude it.

Reply:
done. Please review the latest commit.